sdk/servicebus/Azure.Messaging.ServiceBus/tests/RuleManager/RuleManagerLiveTests.cs

```diff
@@ -0,0 +1,588 @@
+                Assert.True(rulesDescription.Count() == 1);
+                Assert.AreEqual(RuleDescription.DefaultRuleName, rulesDescription.First().Name);
+
+                await ruleManager.RemoveRuleAsync(RuleDescription.DefaultRuleName);
+
+                await ruleManager.AddRuleAsync(new RuleDescription
+                {
+                    Filter = new SqlFilter("Color = 'blue' or Quantity = 10"),
+                    Name = "SqlRuleUsingOperator"
+                });
+
+                rulesDescription = await ruleManager.GetRulesAsync();
+                Assert.True(rulesDescription.Count() == 1);
+                Assert.AreEqual("SqlRuleUsingOperator", rulesDescription.First().Name);
+
+                await SendMessages(client, scope.TopicName);
+
+                var expectedOrders = Orders.Where(c => c.Color == "blue" || c.Quantity == 10).ToList();
+                await ReceiveMessages(
+                    client,
+                    scope.TopicName,
+                    scope.SubscriptionNames.First(),
+                    expectedOrders);
+            }
+        }
+
+        private Order[] Orders = new[]
+            {
+                new Order { Color = "blue", Quantity = 5, Priority = "low" },
+                new Order { Color = "red", Quantity = 10, Priority = "high" },
+                new Order { Color = "yellow", Quantity = 5, Priority = "low" },
+                new Order { Color = "blue", Quantity = 10, Priority = "low" },
+                new Order { Color = "blue", Quantity = 5, Priority = "high" },
+                new Order { Color = "blue", Quantity = 10, Priority = "low" },
+                new Order { Color = "red", Quantity = 5, Priority = "low" },
+                new Order { Color = "red", Quantity = 10, Priority = "low" },
+                new Order { Color = "red", Quantity = 5, Priority = "low" },
+                new Order { Color = "yellow", Quantity = 10, Priority = "high" },
+                new Order { Color = "yellow", Quantity = 5, Priority = "low" },
+                new Order { Color = "yellow", Quantity = 10, Priority = "low" }
+            };
+
+        private async Task SendMessages(ServiceBusClient client, string topicName)
+        {
+            ServiceBusSender sender = client.CreateSender(topicName);
+
+            for (int i = 0; i < Orders.Length; i++)
+            {
+                var message = new ServiceBusMessage(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(Orders[i])))
+                {
+                    CorrelationId = Orders[i].Priority,
+                    Label = Orders[i].Color,
+                    Properties =
+                {
+                    { "color", Orders[i].Color },
+                    { "quantity", Orders[i].Quantity },
+                    { "priority", Orders[i].Priority }
+                }
+                };
+                await sender.SendAsync(message);
+            }
+        }
+
+        private async Task ReceiveMessages(
```

Comment:
nit: name something like ReceiveAndAssertMessages so it is clear that this is doing the assertions.